### PR TITLE
Add a test to verify `nest_trials` for `MLflowCallback` works properly.

### PR DESCRIPTION
**Pull request by harupy**
## Motivation

A follow-up for #1918 

## Description of the changes

Add a test to verify `nest_trials` for `MLflowCallback` works properly.

### CODE REVIEW

File: tests/integration_tests/test_mlflow.py [hunk 1]
```diff
@@ -132,3 +134,31 @@ def test_tag_truncation(tmpdir: py.path.local) -> None:
...
+
+
+def test_nest_trials(tmpdir: py.path.local) -> None:
+    tmp_tracking_uri = "file:{}".format(tmpdir)
+
+    study_name = "my_study"
+    mlflow.set_tracking_uri(tmp_tracking_uri)
+    mlflow.set_experiment(study_name)
+
+    mlflc = MLflowCallback(tracking_uri=tmp_tracking_uri, nest_trials=True)
```
Comment (hvy) [inline tests/integration_tests/test_mlflow.py:146]: We might want to check the negative case where the parameter is `False`.

**Review by hvy — COMMENTED** [review]
Thanks as always. Again, left a nitpick.

Maybe @drobison00 would like to have to a look/review this PR?

File: tests/integration_tests/test_mlflow.py (hunk 1 above)
Comment (harupy) [reply tests/integration_tests/test_mlflow.py:146]: If we set `nest_trials` to `False` and an active run exists, MLflow would raise the following error:

```
Exception: Run with UUID aa67b134da7c43158015648fc00ffdb4 is already active.
To start a new run, first end the current run with mlflow.end_run().
To start a nested run call start_run with nested=True
```

Should I add a test to verify this behavior?

Comment (hvy) [reply tests/integration_tests/test_mlflow.py:146]: Yup, that' what I meant. Sorry I was probably not clear enough.

**Review by hvy — APPROVED** [review]
LGTM!

**Comment by drobison00** [comment]
@hvy Sorry, was out for a few days. LGTM.

**Comment by hvy** [comment]
That's great, let me have this PR merged then!

**Comment by hvy** [comment]
@harupy by the way, you can omit the punctuation from the PR title now. We slightly changed the rules in https://github.com/optuna/optuna/blob/master/CONTRIBUTING.md#title (it used to require a punctuation at the end of the title but not anymore)

**Pull request merged**